File: AlgoExpert/Arrays/Easy/Non constructible change.java
import java.util.*;

class Program {

  public int nonConstructibleChange(int[] coins) {
    if(coins.length == 0) return 1;
		
		Arrays.sort(coins);
		
		int currentChangeCreated = 0;
		
		for(int coin: coins){
			if(coin > currentChangeCreated + 1){
				return currentChangeCreated + 1;
			}
			
			currentChangeCreated += coin;
		}
		
		return currentChangeCreated + 1;
  }
}

File: Python/AlgoExpert/Arrays/Easy/non_constructible_change.py
def nonConstructibleChange(coins):
    # Write your code here.
    coins.sort()
    max_constructible_change = 0

    for coin in coins:
        if coin > max_constructible_change + 1:
            return max_constructible_change + 1
        else:
            max_constructible_change += coin
    
    return max_constructible_change + 1
